File: dotnet/01-commands/tests/ComplexMethod/ComplexMethod/Blacklist.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;

namespace ComplexMethod
{
    public static class Blacklist
    {
        private static List<string> blackListedIP =
        [
            "12.5.6.7",
            "192.0.5.1"
        ];

        public static string GetCurrentIpAddress()
        {
            string hostName = Dns.GetHostName();
            Console.WriteLine(hostName);

            // Get the IP from GetHostByName method of dns class.
            return Dns.GetHostEntry(hostName).AddressList[0].ToString();
        }

        internal static bool IsBlacklisted(string key)
        {
            return blackListedIP.Contains(key);
        }

        internal static bool IsWhiteList(string key)
        {
            return key == GetCurrentIpAddress();
        }
    }
}

File: java/01-commands/tests/sourcecode/src/main/java/com/training/app/ComplexMethod/Blacklist.java
package com.training.app.ComplexMethod;

import java.net.InetAddress;
import java.net.UnknownHostException;
import java.util.ArrayList;
import java.util.List;

public class Blacklist {
    private static List<String> blackListedIP = new ArrayList<>();

    static {
        blackListedIP.add("12.5.6.7");
        blackListedIP.add("192.0.5.1");
    }

    public static String getCurrentIpAddress() throws UnknownHostException {
        InetAddress localhost = InetAddress.getLocalHost();
        return localhost.getHostAddress();
    }

    public static boolean isBlacklisted(String key) {
        return blackListedIP.contains(key);
    }

    public static boolean isWhiteList(String key) throws UnknownHostException {
        return key.equals(getCurrentIpAddress());
    }
}
